src/table.rs

```diff
@@ -572,6 +597,9 @@ impl ToTokens for TableComponentDeriveInput {
 
                                                 let selected_signal = Signal::derive(cx, move || is_sel(Some(item.#key_field.clone())));
 
+                                                // required to be able to get `item` into on_cell_change
+                                                let (row_state, _) = create_signal(cx, item.clone());
```

Comment:
I just remembered that there is a leptos primitive made exactly for this: 
https://docs.rs/leptos/latest/leptos/fn.store_value.html

This saves the reactivity overhead.